File: SuperiorPackGroup/Reports/ItemTransactionXtraReport.vb
Public Class ItemTransactionXtraReport

    Private Sub balacneDetailXrLabel_SummaryCalculated(ByVal sender As System.Object, ByVal e As DevExpress.XtraReports.UI.TextFormatEventArgs) Handles balacneDetailXrLabel.SummaryCalculated

        e.Text = Format(CDbl(e.Value) + CDbl(Me.openingBalanceParameter.Value), "#,##0.00")

    End Sub

    Private Sub ItemTransactionXtraReport_BeforePrint(ByVal sender As Object, ByVal e As Printing.PrintEventArgs) Handles Me.BeforePrint

        If Me.DataSource IsNot Nothing Then
            Me.openingBalanceParameter.Value = CType(Me.DataSource, ItemTransactionsList).OpeningBalance
        End If

    End Sub

End Class

File: SuperiorPackGroup/Reports/ItemTransactionXtraReport.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

namespace SuperiorPackGroup
{
	public partial class ItemTransactionXtraReport
	{

		public ItemTransactionXtraReport()
		{
			InitializeComponent();
		}

		private void balacneDetailXrLabel_SummaryCalculated(System.Object sender, DevExpress.XtraReports.UI.TextFormatEventArgs e)
		{

			e.Text = (Convert.ToDouble(e.Value) + Convert.ToDouble(openingBalanceParameter.Value)).ToString("#,##0.00");

		}

		private void ItemTransactionXtraReport_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e)
		{

			if (DataSource != null)
			{
				openingBalanceParameter.Value = ((ItemTransactionsList)DataSource).OpeningBalance;
			}

		}

	}
}